### DISCUSSION TITLE
Using Bleak in EXE App on Windows 11

### DISCUSSION THREAD

**Discussion by chadcwilliams** [hbldh/bleak#1555] [Q&A] [answered]
As is now apparent from [this](https://github.com/hbldh/bleak/issues/1262) and related issues, Bleak doesn't work on Windows 11 when the `pywin32` and  `pkg_resources` modules are installed. This has been identified as a [bug in the Windows 11 OS](https://github.com/hbldh/bleak/issues/1262#issuecomment-2041601982). Unfortunately, I would like to convert my project into a Windows executable via modules such as `pyinstaller` or `py2exe` but these require those packages. Does anyone know of a workaround to this issue that would make this possible?

I am not familiar with the process of converting python code into executable (or the engine of how bluetooth in Win11 works), so maybe these will always be required packages. If not, I am hoping someone might know of a py to exe converter package that approaches it without these packages, or potentially a method of circumventing the Windows 11 bug? 

Thanks for any comment, even if it's to say that there's nothing I can do.

EDIT: For my use-case, it turns out that removing `pywin32` worked. I have `setuptools` (which recruits `pkg_resources`) installed via `pyinstaller` but that seemed not to be a problem.

**Comment by dlech** [comment, marked as answer]
I totally forgot about the `pywin32` issue https://github.com/hbldh/bleak/issues/1132!. Thanks for bringing it up.

I just made https://github.com/hbldh/bleak/pull/1556 which should now detect that something is wrong and raise and exception before hanging forever. And it includes another possible workaround besides removing `pywin32` in case that isn't an option for some applications.

**Reply by chadcwilliams** [reply]
Amazing work, well done! Keep it up and thanks for all of your effort!
